packages/shared/src/components/streak/popup/ReadingStreakPopup.tsx

```diff
@@ -33,52 +60,52 @@ export function ReadingStreakPopup({
   fullWidth,
 }: ReadingStreakPopupProps): ReactElement {
   const { user } = useAuthContext();
+  const [today, setToday] = useState(new Date());
```

Comment:
Does this need to be state? I think you can just always instantiate it as `const today = new Date()` inside `useMemo`.

Reply:
The `useMemo` called is memoizing based on the value of `today`. I could change it to memoize based on just the date portion of today, but not sure if it would actually work. Let me try :)

Reply:
OK, refactored into single `useMemo`, removed the `setState` hook on `today` and hopefully now the code is more readable.

Added a comment though to the `useMemo`, since now we need to explicitly omit the `today` dependency from it, otherwise it would recalculate on every render.